Marlin/Configuration.h

```diff
@@ -1176,10 +1176,50 @@
 //#define FILAMENT_RUNOUT_SENSOR
 #if ENABLED(FILAMENT_RUNOUT_SENSOR)
   #define FIL_RUNOUT_ENABLED_DEFAULT true // Enable the sensor on startup. Override with M412 followed by M500.
-  #define NUM_RUNOUT_SENSORS   1          // Number of sensors, up to one per extruder. Define a FIL_RUNOUT#_PIN for each.
+  #define NUM_RUNOUT_SENSORS   1          // Number of sensors, up to one per extruder. Define a FIL_RUNOUT#_PIN, and optionally a FIL_RUNOUT#_STATE/FIL_RUNOUT#_PULLUP/FIL_RUNOUT#_PULLDOWN, for each.
   #define FIL_RUNOUT_STATE     LOW        // Pin state indicating that filament is NOT present.
   #define FIL_RUNOUT_PULLUP               // Use internal pullup for filament runout pins.
   //#define FIL_RUNOUT_PULLDOWN           // Use internal pulldown for filament runout pins.
+  /**
+   * With this option each sensor can have a separate state than the first one.
+   * To override the state for a sensor, up to NUM_RUNOUT_SENSORS, set
+   * FIL_RUNOUT#_STATE, otherwise FIL_RUNOUT_STATE will be used. Similarly with
+   * FIL_RUNOUT#_PULLUP and FIL_RUNOUT#_PULLDOWN.
+   */
+  //#define DISTINCT_FIL_RUNOUT_STATES
+  #ifdef DISTINCT_FIL_RUNOUT_STATES
+    //#define FIL_RUNOUT1_STATE LOW
```

Comment:
We only need a single pull setting because we know whether the pull will be up or down based on the runout state. See `POWER_LOSS_PULL` for example.

Reply:
So `POWER_LOSS_PULL` does set the pullup/pulldown based on the value of state, but for runout we actually [don't do that](https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/Marlin/src/feature/runout.h#L152-L158):
```cpp
#if ENABLED(FIL_RUNOUT_PULLUP)
  #define INIT_RUNOUT_PIN(P) SET_INPUT_PULLUP(P)
#elif ENABLED(FIL_RUNOUT_PULLDOWN)
  #define INIT_RUNOUT_PIN(P) SET_INPUT_PULLDOWN(P)
#else
  #define INIT_RUNOUT_PIN(P) SET_INPUT(P)
#endif
```

I could convert it to be like power loss if you want me:
```cpp

#if ENABLED(FIL_RUNOUT_PULL)
  #if FIL_RUNOUT_STATE == LOW
    #define INIT_RUNOUT_PIN(P) SET_INPUT_PULLUP(P)
  #else
    #define INIT_RUNOUT_PIN(P) SET_INPUT_PULLDOWN(P)
  #endif
#else
  #define INIT_RUNOUT_PIN(P) SET_INPUT(P)
#endif
```

Reply:
@thinkyhead Let me know if you want me to convert `FIL_RUNUOUT_PULLUP`/`FIL_RUNOUT_PULLDOWN` to `FIL_RUNOUT_PULL`

Reply:
Sorry for the slow reply. Lots of PRs coming through lately. Anyway, yes, it would be preferable for the filament runout pin init to work like the power loss pin init. That helps keep the number of options minimal.